proposals/char-kind-proposal.md

```diff
@@ -0,0 +1,173 @@
+
+First of all, we overview the additional type families implemented by us in this patch.
+
+```haskell
+type family CmpChar (a :: Char) (b :: Char) :: Ordering
+```
+
+Comparison of type-level characters, as a type family. A type-level analogue of the function `compare` specified for characters.
+
+```haskell
+type family LeqChar (a :: Char) (b :: Char) :: Bool
+```
+
+This is a type-level comparison of characters as well. `LeqChar` yields a Boolean value and corresponds to `(<=)`.
+
+```haskell
+type family ConsSymbol (a :: Char) (b :: Symbol) :: Symbol
+```
+
+This extends a type-level symbol with a type-level character
+
+```haskell
+type family UnconsSymbol (a :: Symbol) :: Maybe (Char, Symbol)
+```
+
+This type family yields type-level `Just` storing the first character of a symbol and its tail if it is nonempty and `Nothing` otherwise.
+
+
+Type-level counterparts of the functions `toUpper`, `toLower`, and `toTitle` from 'Data.Char'.
+
+```haskell
+type family ToUpper (a :: Char) :: Char
+
+type family ToLower (a :: Char) :: Char
+
+type family ToTitle (a :: Char) :: Char
+```
+
+These type families are type-level analogues of the functions `ord` and `chr` from Data.Char respectively.
+
+```haskell
+type family CharToNat (a :: Char) :: Nat
+
+type family NatToChar (a :: Nat) :: Char
+```
+
+A type-level analogue of the function `generalCategory` from `Data.Kind`.
+
+```haskell
+type family GeneralCharCategory (a :: Char) :: GeneralCategory
+```
+
+
+The second group of type families consists of built-in unary predicates. All of them are based on their corresponding term-level analogues from `Data.Char`. The precise list is the following one:
+
+```haskell
+type family IsAlpha (a :: Char) :: Bool
+
+type family IsAlphaNum (a :: Char) :: Bool
+
+type family IsControl (a :: Char) :: Bool
+
+type family IsPrint (a :: Char) :: Bool
+
+type family IsUpper (a :: Char) :: Bool
+
+type family IsLower (a :: Char) :: Bool
+
+type family IsSpace (a :: Char) :: Bool
+
+type family IsDigit (a :: Char) :: Bool
+
+type family IsOctDigit (a :: Char) :: Bool
+
+type family IsHexDigit (a :: Char) :: Bool
+
+type family IsLetter (a :: Char) :: Bool
+```
+
+We also provide several type-level predicates implemented via the `GeneralCharCategory` type family.
+
+```haskell
+type IsMark a = IsMarkCategory (GeneralCharCategory a)
+
+type IsNumber a = IsNumberCategory (GeneralCharCategory a)
+
+type IsPunctuation a = IsPunctuationCategory (GeneralCharCategory a)
+
+type IsSymbol a = IsSymbolCategory (GeneralCharCategory a)
+
+type IsSeparator a = IsSeparatorCategory (GeneralCharCategory a)
+```
+
+Built-in type families we described above are supported with the corresponding definitions and functions in the modules `compiler/GHC/Builtin/Names.hs`, `compiler/GHC/Builtin/Types.hs`, and `compiler/GHC/Builtin/Types/Literals.hs`. In addition to type families, our patch contain the following updates:
+
+1. parsing the 'x' syntax
```

Comment:
This list is not a valid specification.

* Why does it include points like "Haddock-related updates" and "tests"?
* What does it mean to have "template-haskell support", specifically, which constructors are added to the TH AST?
* Points such as "parsing the 'x' syntax" require more context/elaboration, e.g. "Allow the character literal syntax in type-level contexts (e.g. `f :: Proxy 'x'`)."

In other words, please rewrite the entire list in such a way that it forms a specification, rather than a todo/checkpoint list.